### PR TITLE
updated docs

### PR DESCRIPTION
**Pull request by wilkermichael**
- added docs for start command
- deprecated running without a command
- added instructions for autocomplete setup

Start Command: https://consul-6ewood6ws-hashicorp.vercel.app/docs/nia/cli/start
No Command Deprecation Daemon: https://consul-6ewood6ws-hashicorp.vercel.app/docs/nia/cli

### CODE REVIEW

File: website/content/docs/nia/cli/index.mdx [hunk 1]
```diff
@@ -11,6 +11,8 @@ Consul-Terraform-Sync (CTS) is controlled via an easy to use command-line interf
 
 ## Daemon
 
+~> Running daemon mode without a command is deprecated in CTS 0.6.0 and will be removed at a much later date in a major release. For information on the preferred way to run daemon mode see the [`start` command docs](/docs/nia/cli/start)
```
Comment (wilkermichael) [inline website/content/docs/nia/cli/index.mdx:14]: @devarshishah3 can you please review the phrasing here?

File: website/content/docs/nia/cli/start.mdx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Behavior: This is the default mode in which CTS passes through a once-mode phase and then turns into a long running process. During the once-mode phase, the daemon will exit with a non-zero status if it encounters an error. After successfully passing through once-mode phase, it will begin a long-running process in which errors are logged and exiting is not expected behavior. Once beginning the long-running process, the daemon serves any API and command requests.
+
+#### Inspect Mode
+
+Flag: `-inspect`
+
+Behavior: CTS will display the proposed state changes for all tasks once and exit. No changes are applied in this mode. On encountering an error before completing, CTS will exit with a non-zero status.
+
+Usage: Intended to be run before daemon-mode in order to confirm configuration is accurate and tasks would update network infrastructure as expected.
```
Comment (lornasong) [inline website/content/docs/nia/cli/start.mdx:34]: If I understand correctly, "daemon-mode" here should be "long-running mode". Want to acknowledge that this language was pre-existing from the index page.

File: website/content/docs/nia/cli/start.mdx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+Usage: Useful to debug one or more tasks to confirm configuration is accurate and the selected tasks would update network infrastructure as expected.
+
+#### Once Mode
+
+Flag: `-once`
+
+Behavior: CTS will run all tasks once with buffer periods disabled and exit. On encountering an error before completing, CTS will exit with a non-zero status.
+
+Usage: Intended to be run before daemon-mode in order to confirm configuration is accurate and tasks update network infrastructure as expected.
```
Comment (lornasong) [inline website/content/docs/nia/cli/start.mdx:50]: Similarly here, is it more accurate to replace "daemon-mode" with "long-running mode"? Acknowledging this language pre-existed this PR.

File: website/content/docs/nia/cli/start.mdx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+---
+layout: docs
+page_title: Start Command
+description: >-
+  Consul-Terraform-Sync supports start command for running CTS in daemon mode
+---
+
+# start
+
+The start command runs Consul-Terraform-Sync (CTS) in Daemon mode. When running as a daemon, there is no default configuration to run CTS in a meaningful way. Setting a configuration flag `-config-file` or `-config-dir` is required. For example:
```
Comment (lornasong) [inline website/content/docs/nia/cli/start.mdx:10]:
```suggestion
The `start` command starts Consul-Terraform-Sync (CTS) in daemon mode. When running as a daemon, there is no default configuration to start CTS in a meaningful way. Setting a configuration flag `-config-file` or `-config-dir` is required. For example:
```

Feel free to take / leave ^. The snippet suggests
 - adding "``" around "start" in "start command" - aligns with other command docs
 - lower casing "d" in "Daemon mode"
 - switches to using "start" language to align with name of command

Comment (lornasong) [reply website/content/docs/nia/cli/start.mdx:10]: General suggestion to search on "run" and replace (where it makes sense) with "start" to add alignment with name of command

File: website/content/docs/nia/cli/start.mdx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+Usage: Intended to be run before daemon-mode in order to confirm configuration is accurate and tasks update network infrastructure as expected.
+
+### Options
+
+The `start` command supports the following options:
+
+| Name            | Required | Type    | Description                     |Default                  |
+| --------------- | -------- | ------- | ------------------------------- | ----------------------- |
+| `-config-dir` | Optional | string | A directory to load files for configuring CTS. Configuration files require an .hcl or .json file extension in order to specify their format. This option can be specified multiple times to load different directories. | none |
```
Comment (lornasong) [inline website/content/docs/nia/cli/start.mdx:58]: I noticed that the flags in the "Name" column wrapped around in the preview. FYI @mkam found a neat workaround that might be useful here: https://github.com/hashicorp/consul/pull/12371#discussion_r808512456

File: website/content/docs/nia/cli/start.mdx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+| `-config-file` | Optional | string | A file to load for configuring CTS. Configuration file requires an .hcl or .json extension in order to specify their format. This option can be specified multiple times to load different configuration files. | none |
+| `-inspect` | Optional | boolean | Run CTS in Inspect mode to print the proposed state changes for all tasks, and then exit. No changes are applied in this mode. | false |
+| `-inspect-task` | Optional | string | Run CTS in Inspect mode to print the proposed state changes for the task, and then exit. No changes are applied in this mode. | none |
```
Comment (lornasong) [inline website/content/docs/nia/cli/start.mdx:61]: It's already mentioned earlier, but what are your thoughts on adding the "This option can be specified multiple times..." type of line to `-inspect-task` as well?

File: website/content/docs/nia/cli/start.mdx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+| `-once` | Optional | boolean | Render templates and run tasks once. Does not run the process as a daemon and disables buffer periods. | false |
```
Comment (lornasong) [inline website/content/docs/nia/cli/start.mdx:62]: Re: "does not run the process as a daemon": this is related to earlier "daemon" language questions - want to double-check if once mode is a sub-type of daemon mode?

**Review by lornasong — COMMENTED** [review]
Excited about autocompletion and start cli! I had some questions around the "daemon" language that I left as comments. The comments were the clearest way for me to get across my thoughts, but if they're a larger conversation happy to chat about anything in whatever way works best

File: website/content/docs/nia/cli/index.mdx (hunk 1 above)
Comment (devarshishah3) [reply website/content/docs/nia/cli/index.mdx:14]: Looks good!

**Review by devarshishah3 — APPROVED** [review]

File: website/content/docs/nia/cli/index.mdx [hunk 8]
```diff
@@ -11,6 +11,8 @@ Consul-Terraform-Sync (CTS) is controlled via an easy to use command-line interf
...
+
 When running as a daemon, there is no default configuration to run CTS in a meaningful way. Setting a configuration flag `-config-file` or `-config-dir` is required. For example:
 
 ```shell-session
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply website/content/docs/nia/cli/index.mdx:18]: yeah, I think that makes sense, I'll remove it

File: website/content/docs/nia/cli/start.mdx (hunk 4 above)
Comment (wilkermichael) [reply website/content/docs/nia/cli/start.mdx:10]: that's a good point, I'll do that

File: website/content/docs/nia/cli/start.mdx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+```shell-session
+$ consul-terraform-sync start -config-file=config.hcl
+```
+
+To view a list of available flags, use the `-help` or `-h` flag.
+
+### Modes
+
+CTS can be run as a daemon in different modes.
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply website/content/docs/nia/cli/start.mdx:20]: yeah, so have I... but isn't a daemon just a background process? I wouldn't call once-mode daemon mode

File: website/content/docs/nia/cli/start.mdx (hunk 2 above)
Comment (wilkermichael) [reply website/content/docs/nia/cli/start.mdx:34]: yeah, this is from the pre-existing information

File: website/content/docs/nia/cli/index.mdx [hunk 10]
```diff
@@ -19,28 +21,6 @@ $ consul-terraform-sync -config-file=config.hcl
...
-#### Inspect Mode
-
-Flag: `-inspect`
-
-Behavior: CTS will display the proposed state changes for all tasks once and exit. No changes are applied in this mode. On encountering an error before completing, CTS will exit with a non-zero status.
-
-Usage: Intended to be run before daemon-mode in order to confirm configuration is accurate and tasks would update network infrastructure as expected.
-
----
-
```
Comment (lornasong) [inline website/content/docs/nia/cli/index.mdx:43]: Looks like you can remove more? - including Once Mode (up to the Command section)

**Review by lornasong — APPROVED** [review]

**Pull request merged**